test/test_torch.py

```diff
@@ -646,7 +646,8 @@ def cpp_warn_fn():
             warning = w[0]
 
             # Checks for cpp context in the warning message
-            self.assertTrue(re.search(s, str(warning.message)) is not None)
+            escaped_warning_message = str(warning.message).encode('unicode_escape')
+            self.assertTrue(re.search(s, str(escaped_warning_message), re.IGNORECASE) is not None)
```

Comment:
What exactly is going on here? Why is it needed for fbcode but not OSS?

Reply:
@esauser2, I'm also wondering why this is needed here. I'm trying to make changes to this test, but it's not clear to me why `encode('unicode_escape')` is needed. The only way I've been able to replicate a failure is to include a newline in the path to `RangeFactories.cpp` using `str.replace`, and that situation could be more cleanly solved using [`re.MULTILINE`](https://docs.python.org/3.10/library/re.html#re.MULTILINE). Is there something special about your working directory path that causes it to have special characters or encodings?

Reply:
IIRC, I think this came from the path `...\native\rangefactories.cpp` (Windows machine => case insensitive and using backslashes instead of slashes), and I figured out that `\r` had to be escaped

Reply:
That's good to know! The change I made in cb2b7b1e57e965493e14f465de869a78f50967cb shouldn't break that.